unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCrypto.tsx

```diff
@@ -195,20 +196,52 @@ export function ConfirmCrypto({
       }
 
       const walletService = await getWalletService(lockNetwork)
-      await walletService.purchaseKeys(
-        {
-          lockAddress,
-          keyPrices,
-          owners: recipients!,
-          data: purchaseData,
-          keyManagers: keyManagers?.length ? keyManagers : undefined,
-          recurringPayments,
-          referrers,
-          totalApproval,
-        },
-        {} /** Transaction params */,
-        onErrorCallback
-      )
+      if (renew) {
+        if (lock!.publicLockVersion! <= 9) {
```

Comment:
I honestly think we should had a renew function to unlockjs that would retrieve the token Id if needed
Because you will have the same logic on locksmith